airflow/contrib/executors/kubernetes_executor.py

```diff
@@ -252,6 +265,15 @@ def _validate(self):
                 'or `dags_in_image` '
                 'or `git_repo and git_branch and git_dags_folder_mount_point`')
 
+        # cannot allow certain labels, as airflow needs them
+        if self.kube_extra_labels:
+            for k in self.protected_labels:
+                if k in self.kube_extra_labels:
+                    raise AirflowConfigException(
+                        'In kubernetes mode `extra_labels` cannot include '
+                        '`airflow-worker`, `dag_id`, `task_id`, `try_number`, or `execution_date`'
```

Comment:
I prefer to use `self.protected_labels` instead of hardcode it here ^

Reply:
I agree with this.